### ISSUE TITLE
Editorconfig rules fail to apply on Windows 10

### ISSUE THREAD

**Issue by viveleroi** [evanshortiss/lintspaces-cli#25]
On Windows (7 and 10) the `-e` editorconfig flag isn't working. The rules which should be applied, aren't triggering any errors. When we try with the argument versions (like `-t`) it catches errors.

However, this works fine on macOS.

In `package.json`, we have a scripts entry called `lint-spaces`, which uses the following command:

`node ./node_modules/lintspaces-cli/index.js -i java-comments -e .editorconfig -. *.{xml,yml,json}`

And here's our editorconfig:

```
# top-most EditorConfig file
root = true

[*]
charset = utf-8
end_of_line = lf
indent_size = 4
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true

[package.json]
indent_size = 2
```

**Comment by viveleroi** [comment]
Trying that, it's definitely not a path/resolution issue. I *don't* see the error that would be triggered on line 44 and the function returns the correct path. So it's finding the file, but the rules aren't used.

**Comment by evanshortiss** [comment]
Sounds like it might be a bug in the core lintspaces module since this is just a wrapper. Could maybe be a breaking change, or as you suggest something Windows specific, again in the core since it sounds like the file is being found and passed along, but I can check later this week on my Windows machine.

**Comment by viveleroi** [comment]
I'd appreciate hearing whatever you find out. I've reverted to using the command-line arguments for now, though it doesn't appear like those support per-file rules, for example 2 spaces for package.json, but 4 for others, etc?

**Comment by schorfES** [comment]
Hello @viveleroi, I assume that your issue is related to the same issue that was reported in the lintspaces core module (https://github.com/schorfES/node-lintspaces/issues/40). It's related to how editorconfig traverses and resolves the path on windows machines. 

It's released in version [0.8.0 of lintspaces](https://github.com/schorfES/node-lintspaces/releases/tag/0.8.0) – we need to catchup the dependency version.

**Comment by viveleroi** [comment]
Has this been resolved yet? I'm now setting this plugin up in a new project on Windows 11 and am running into this exact same problem. I really don't want to hard-code all of the same rules as arguments.

**Comment by evanshortiss** [comment]
@viveleroi can you test the `update-lintspaces-for-windows` branch? `npm install evanshortiss/lintspaces-cli#update-lintspaces-for-windows`

If it works I can merge and do a release.

**Comment by viveleroi** [comment]
Seems to work, although only if I specifically pass the file name of the editorconfig:

`lintspaces -e .editorconfig *.{html,json} .*.yml`

Without the `.editorconfig` bit it didn't show any errors, despite having an invalid test file. The errors do show when I include the filename.

Considering the file name is always the same, it would be nice if it defaults to it. Or even better, if it detects the editorconfig file and doesn't need the `-e` argument unless other args were passed.

**Comment by evanshortiss** [comment]
Thanks for that. Noted and fixed about the `-e` parameter. That will now default to use the editorconfig file in the current working directory. Released as v0.8.0.
